imagetest/test_suites/storageperf/setup.go

```diff
@@ -29,17 +29,24 @@ func TestSetup(t *imagetest.TestWorkflow) error {
 		return fmt.Errorf("boot disk and mount disk must be different sizes for disk identification")
 	}
 	// initialize vm with the hard coded machine type and platform corresponding to the index
-	paramMaps := []map[string]string{
-		{"machineType": "c3-standard-88", "diskType": imagetest.HyperdiskExtreme, "metric": "C3_CPUS"},
-		// temporarily disable c3d hyperdisk until the api allows it again
-		// {"machineType": "c3d-standard-180", "zone": "us-east4-c", "diskType": imagetest.HyperdiskExtreme, "metric": "CPUS"}, // No public metric for this yet but the CPU count will work because they're so large
-		{"machineType": "n2-standard-80", "diskType": imagetest.HyperdiskExtreme, "metric": "N2_CPUS"},
-		{"machineType": "c3-standard-88", "diskType": imagetest.PdBalanced, "metric": "C3_CPUS"},
-		{"machineType": "c3d-standard-180", "zone": "us-east4-c", "diskType": imagetest.PdBalanced, "metric": "CPUS"},
-		{"machineType": "n2d-standard-64", "diskType": imagetest.PdBalanced, "metric": "N2D_CPUS"},
-		{"machineType": "n1-standard-64", "diskType": imagetest.PdBalanced, "minCpuPlatform": "Intel Skylake", "metric": "CPUS"}, // N1 CPUS aren't measured separately
-		// zone for h3 is a temporary measure while h3 is pre-GA
-		{"machineType": "h3-standard-88", "zone": "us-central1-a", "diskType": imagetest.PdBalanced, "metric": "CPUS"}, // No public metric
+	var paramMaps []map[string]string
+	if strings.Contains(t.Image, "arm64") {
```

Comment:
What if we do it a bit more robust and make `paramMap` a bit smarter and give it the knowledge of what's the target os type? We can either make `paramMaps` like `[]TestConfig` where `TestConfig` (maybe `TestConfig` is not the best naming though) would be:

```golang
type TestConfig struct {
   machineType string
   diskType string
   metric string
   targetOS string
   ...
}
```

And then when using `paramMaps` we'd filter it out based on the required criteria, being it target OS or anything else depending on the requirements we have in hand.

Reply:
I'd like to wait until after #800 is reviewed to do something like that, it would be much more robust (and easier) to use the compute API directly to do this. I can hold this PR until then.

Reply:
Take a look, switched to a struct with defined fields for existing options and handled each of them in a fairly similar way to what's being done currently.